File: dataset/ZeroShot - Copy/CLAUDE_claude-sonnet-4-5-20250929/Task67.c

#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <sqlite3.h>

typedef struct {
    char username[100];
    char name[100];
    char email[100];
    char phone[20];
} Customer;

Customer* getCustomerByUsername(const char* customerUsername) {
    if (customerUsername == NULL || strlen(customerUsername) == 0) {
        return NULL;
    }
    
    sqlite3* db;
    sqlite3_stmt* stmt;
    const char* dbPath = "customer.db";
    
    if (sqlite3_open(dbPath, &db) != SQLITE_OK) {
        fprintf(stderr, "Database error: %s\\n", sqlite3_errmsg(db));
        return NULL;
    }
    
    const char* query = "SELECT username, name, email, phone FROM customer WHERE username = ?";
    
    if (sqlite3_prepare_v2(db, query, -1, &stmt, NULL) != SQLITE_OK) {
        fprintf(stderr, "Query preparation error: %s\\n", sqlite3_errmsg(db));
        sqlite3_close(db);
        return NULL;
    }
    
    sqlite3_bind_text(stmt, 1, customerUsername, -1, SQLITE_TRANSIENT);
    
    Customer* customer = NULL;
    
    if (sqlite3_step(stmt) == SQLITE_ROW) {
        customer = (Customer*)malloc(sizeof(Customer));
        if (customer != NULL) {
            strncpy(customer->username, (const char*)sqlite3_column_text(stmt, 0), 99);
            customer->username[99] = '\\0';
            strncpy(customer->name, (const char*)sqlite3_column_text(stmt, 1), 99);
            customer->name[99] = '\\0';
            strncpy(customer->email, (const char*)sqlite3_column_text(stmt, 2), 99);
            customer->email[99] = '\\0';
            strncpy(customer->phone, (const char*)sqlite3_column_text(stmt, 3), 19);
            customer->phone[19] = '\\0';
        }
    }
    
    sqlite3_finalize(stmt);
    sqlite3_close(db);
    
    return customer;
}

void initializeDatabase() {
    sqlite3* db;
    char* errMsg = NULL;
    
    if (sqlite3_open("customer.db", &db) != SQLITE_OK) {
        fprintf(stderr, "Database initialization error\\n");
        return;
    }
    
    const char* createTable = 
        "CREATE TABLE IF NOT EXISTS customer ("
        "username TEXT PRIMARY KEY, "
        "name TEXT, "
        "email TEXT, "
        "phone TEXT)";
    
    sqlite3_exec(db, createTable, NULL, NULL, &errMsg);
    sqlite3_exec(db, "DELETE FROM customer", NULL, NULL, &errMsg);
    
    sqlite3_exec(db, "INSERT INTO customer VALUES ('john_doe', 'John Doe', 'john@example.com', '555-0101')", NULL, NULL, &errMsg);
    sqlite3_exec(db, "INSERT INTO customer VALUES ('jane_smith', 'Jane Smith', 'jane@example.com', '555-0102')", NULL, NULL, &errMsg);
    sqlite3_exec(db, "INSERT INTO customer VALUES ('bob_wilson', 'Bob Wilson', 'bob@example.com', '555-0103')", NULL, NULL, &errMsg);
    sqlite3_exec(db, "INSERT INTO customer VALUES ('alice_brown', 'Alice Brown', 'alice@example.com', '555-0104')", NULL, NULL, &errMsg);
    sqlite3_exec(db, "INSERT INTO customer VALUES ('charlie_davis', 'Charlie Davis', 'charlie@example.com', '555-0105')", NULL, NULL, &errMsg);
    
    sqlite3_close(db);
}

void printCustomer(const Customer* customer) {
    printf("Customer{username='%s', name='%s', email='%s', phone='%s'}\\n",
           customer->username, customer->name, customer->email, customer->phone);
}

int main(int argc, char* argv[]) {
    initializeDatabase();
    
    const char* testCases[] = {"john_doe", "jane_smith", "bob_wilson", "nonexistent_user", ""};
    
    if (argc > 1) {
        Customer* customer = getCustomerByUsername(argv[1]);
        if (customer != NULL) {
            printf("Found: ");
            printCustomer(customer);
            free(customer);
        } else {
            printf("Customer not found: %s\\n", argv[1]);
        }
    } else {
        printf("Running test cases:\\n");
        for (int i = 0; i < 5; i++) {
            printf("\\nSearching for: '%s'\\n", testCases[i]);
            Customer* customer = getCustomerByUsername(testCases[i]);
            if (customer != NULL) {
                printf("Result: ");
                printCustomer(customer);
                free(customer);
            } else {
                printf("Result: Customer not found\\n");
            }
        }
    }
    
    return 0;
}

File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task67.java

import java.sql.*;
import java.util.*;

class Customer {
    private String username;
    private String name;
    private String email;
    private String phone;
    
    public Customer(String username, String name, String email, String phone) {
        this.username = username;
        this.name = name;
        this.email = email;
        this.phone = phone;
    }
    
    @Override
    public String toString() {
        return "Customer{username='" + username + "', name='" + name + 
               "', email='" + email + "', phone='" + phone + "'}";
    }
}

public class Task67 {
    private static final String DB_URL = "jdbc:sqlite:customer.db";
    
    public static Customer getCustomerByUsername(String customerUsername) {
        if (customerUsername == null || customerUsername.trim().isEmpty()) {
            return null;
        }
        
        String query = "SELECT username, name, email, phone FROM customer WHERE username = ?";
        
        try (Connection conn = DriverManager.getConnection(DB_URL);
             PreparedStatement pstmt = conn.prepareStatement(query)) {
            
            pstmt.setString(1, customerUsername);
            
            try (ResultSet rs = pstmt.executeQuery()) {
                if (rs.next()) {
                    return new Customer(
                        rs.getString("username"),
                        rs.getString("name"),
                        rs.getString("email"),
                        rs.getString("phone")
                    );
                }
            }
        } catch (SQLException e) {
            System.err.println("Database error: " + e.getMessage());
        }
        
        return null;
    }
    
    private static void initializeDatabase() {
        try (Connection conn = DriverManager.getConnection(DB_URL);
             Statement stmt = conn.createStatement()) {
            
            stmt.execute("CREATE TABLE IF NOT EXISTS customer (" +
                        "username TEXT PRIMARY KEY, " +
                        "name TEXT, " +
                        "email TEXT, " +
                        "phone TEXT)");
            
            stmt.execute("DELETE FROM customer");
            
            stmt.execute("INSERT INTO customer VALUES ('john_doe', 'John Doe', 'john@example.com', '555-0101')");
            stmt.execute("INSERT INTO customer VALUES ('jane_smith', 'Jane Smith', 'jane@example.com', '555-0102')");
            stmt.execute("INSERT INTO customer VALUES ('bob_wilson', 'Bob Wilson', 'bob@example.com', '555-0103')");
            stmt.execute("INSERT INTO customer VALUES ('alice_brown', 'Alice Brown', 'alice@example.com', '555-0104')");
            stmt.execute("INSERT INTO customer VALUES ('charlie_davis', 'Charlie Davis', 'charlie@example.com', '555-0105')");
            
        } catch (SQLException e) {
            System.err.println("Database initialization error: " + e.getMessage());
        }
    }
    
    public static void main(String[] args) {
        initializeDatabase();
        
        String[] testCases = {"john_doe", "jane_smith", "bob_wilson", "nonexistent_user", ""};
        
        if (args.length > 0) {
            Customer customer = getCustomerByUsername(args[0]);
            if (customer != null) {
                System.out.println("Found: " + customer);
            } else {
                System.out.println("Customer not found: " + args[0]);
            }
        } else {
            System.out.println("Running test cases:");
            for (String username : testCases) {
                System.out.println("\\nSearching for: '" + username + "'");
                Customer customer = getCustomerByUsername(username);
                if (customer != null) {
                    System.out.println("Result: " + customer);
                } else {
                    System.out.println("Result: Customer not found");
                }
            }
        }
    }
}
